drivers/modem/quectel-bg9x.h

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2020 Analog Life LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef QUECTEL_BG9X_H
+#define QUECTEL_BG9X_H
+
+#include <kernel.h>
+#include <ctype.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <drivers/gpio.h>
+#include <device.h>
+#include <init.h>
+
+#include <net/net_if.h>
+#include <net/net_offload.h>
+#include <net/socket_offload.h>
+
+#include "modem_context.h"
+#include "modem_socket.h"
+#include "modem_cmd_handler.h"
+#include "modem_iface_uart.h"
+
+#define MDM_UART_DEV_NAME		  DT_INST_BUS_LABEL(0)
+#define MDM_CMD_TIMEOUT			  K_SECONDS(10)
+#define MDM_CMD_CONN_TIMEOUT		  K_SECONDS(120)
+#define MDM_REGISTRATION_TIMEOUT	  K_SECONDS(180)
+#define MDM_SENDMSG_SLEEP		  K_MSEC(1)
+#define MDM_MAX_DATA_LENGTH		  1024
+#define MDM_RECV_MAX_BUF		  30
+#define MDM_RECV_BUF_SIZE		  1024
```

Comment:
`NET_BUF_POOL_DEFINE` allocates 30 * 1024 for the `mdm_recv_pool`. Can this be reduced?

Reply:
May be the best solution is to put either of these values in Kconfig ? This will allow users to make modifications based on their needs without modifying the source code, And there's no reason why reducing the size of pool won't work..